rules/exposure-to-internet/raw.rego

```diff
@@ -0,0 +1,75 @@
+package armo_builtins
+
+# Checks if NodePort or LoadBalancer is connected to a workload to expose something
+deny[msga] {
+    service := input[_]
+    service.kind == "Service"
+    is_unsafe_service(service)
+    
+    wl := input[_]
+    spec_template_spec_patterns := {"Deployment", "ReplicaSet", "DaemonSet", "StatefulSet", "Job", "CronJob"}
+    spec_template_spec_patterns[wl.kind]
+    not wl_connected_to_service(wl, service)
+    result := ["spec.selector.matchLabels", "service.spec.selector"]
+    
+    msga := {
+        "alertMessage": sprintf("service: %v is exposed", [service.metadata.name]),
+        "packagename": "armo_builtins",
+        "alertScore": 7,
+        "fixPaths": [],
+        "failedPaths": result,
+        "alertObject": {
+            "k8sApiObjects": [service]
+        }
+    }
+}
+
+# Checks if Ingress is connected to a service and a workload to expose something
+deny[msga] {
+    ingress := input[_]
+    ingress.kind == "Ingress"
+    
+    svc := input[_]
+    svc.kind == "Service"
+    not is_unsafe_service(svc)
+
+    wl := input[_]
+    spec_template_spec_patterns := {"Deployment", "ReplicaSet", "DaemonSet", "StatefulSet", "Job", "CronJob"}
+    spec_template_spec_patterns[wl.kind]
+    result := ["service.spec.selector", "ingress.spec.rules[].http.paths[].backend.serviceName"] 
```

Comment:
We should use the exact paths and not just [], you should use iterators for that.
You can see example for how to use it [here](https://github.com/kubescape/regolibrary/pull/454/files#diff-907da54cc2c57f76b94b9b45d29e95b09f0d30aa57ce302ce137af26f3ca485f).